build/lib/image2excel.py

```diff
@@ -3,35 +3,42 @@
 import os
 from colormap import rgb2hex
 from matplotlib.image import imread
-from PIL import Image
+from PIL import Image, ImageFile
 from openpyxl import load_workbook
+from typing import Union
 
-def __default_path(fileName):
+#ignore missing imports for colormap -> mypy.ini ignore_missing_imports = True
+
+def __default_path(fileName : str) -> str:
     path = os.getcwd()
     file = os.path.join(path,fileName)
     return file
 
-def __colorCells(string):
+def __colorCells(string : str) -> str:
     return 'background-color:' + string
 
-def __changeZoom(excelFile,zoom=10):
+def __changeZoom(excelFile : pd.io.excel._base.ExcelFile, zoom=10):
     wb = load_workbook(excelFile)
     for ws in wb.worksheets:
         ws.sheet_view.zoomScale = zoom
     wb.save(excelFile)
 
-def __resize_picture(file,keep_aspect,extension):
-    image = Image.open(file)
+def __resize_picture(file, keep_aspect, extension) -> Image.Image:
+    image: Union[Image.Image, ImageFile.ImageFile]  = Image.open(file)
     if(sorted(image.size)[0]> 260 or sorted(image.size)[1]> 300):
         if keep_aspect:
-            image.thumbnail(tuple([300/max(image.size)*x for x in image.size]), Image.ANTIALIAS)
+            width, height = image.size
+            new_size = (300 / max(image.size) * width, 300 / max(image.size) * height)
+            image.thumbnail(new_size, Image.Resampling.LANCZOS)
+            #image.thumbnail(tuple([300/max(image.size)*x for x in image.size]), Image.Resampling.LANCZOS) #Image.ANTIALIAS depricated
+            #getting error cause the image can be over 2D
         else:
-            image = image.resize((260,300),Image.ANTIALIAS)
+            image = image.resize((260,300), Image.Resampling.LANCZOS) #Image.ANTIALIAS depricated
         file=file.replace(extension,"_small" + extension)
         image.save(file)
     return file
 
-def im2xlsx(file,resize=True,keep_aspect=False):
+def im2xlsx(file, resize=True, keep_aspect=False):
```

Comment:
@BarbaraLearn we can also and should also add type annotation here, regardless of the function not returning anything, we can use a None annotation for the output